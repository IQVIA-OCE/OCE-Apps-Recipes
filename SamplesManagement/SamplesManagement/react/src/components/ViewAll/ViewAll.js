import React from 'react';
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native';
import { externalNavigator } from '../../../bridge/Navigation/ExternalNavigator';

const navigateToUrl = async url => {
  try {
    await externalNavigator.open(url);
  } catch (error) {
    console.error(error);
  }
};

const ViewAll = ({ url, onPress }) => {
  const handlePress = () => {
    if (url) {
      navigateToUrl(url);
    } else if (typeof onPress === 'function') {
      onPress();
    }
  };
  return (
    <View style={styles.viewAll}>
      <TouchableOpacity onPress={handlePress}>
        <Text style={styles.viewAllText}>View all</Text>
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  viewAll: {
    flexDirection: 'row',
    justifyContent: 'flex-end',
    alignItems: 'flex-start',
    borderTopWidth: 1,
    borderTopColor: '#dddbda',
    paddingTop: 10,
    marginTop: 5,
  },
  viewAllText: {
    color: '#0070d2',
    fontSize: 13,
  },
});

export default ViewAll;
