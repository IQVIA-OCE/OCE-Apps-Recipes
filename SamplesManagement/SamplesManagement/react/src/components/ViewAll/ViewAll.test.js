import React from 'react';
import renderer from 'react-test-renderer';
import ViewAll from './ViewAll';
import { TouchableOpacity } from 'react-native';
import { NativeModules } from 'react-native';

let open;
let onPress = jest.fn()

describe('ViewAll', () => {
  beforeEach(() => {
    open = jest
      .fn()
      .mockResolvedValueOnce('URL')
    NativeModules.ExternalNavigatorBridge = { open };
  });

  it('Should render ViewAll component and redirect to external url', () => {
    const tree = renderer.create(<ViewAll url={'{EndPoint}'} />);

    tree.root.findByType(TouchableOpacity).props.onPress();

    expect(open.mock.calls.length).toBe(1);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('Should render ViewAll component and handle press', () => {
    const tree = renderer.create(<ViewAll onPress={onPress} />);

    tree.root.findByType(TouchableOpacity).props.onPress();

    expect(onPress).toBeCalledTimes(1);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe('ViewAll', () => {
  beforeEach(() => {
    open = jest
      .fn()
      .mockRejectedValueOnce(new Error('Async error'));
    NativeModules.ExternalNavigatorBridge = { open };
  });

  it('Should render ViewAll component and throw an error', async () => {
    const tree = renderer.create(<ViewAll url={'{EndPoint}'} />);

    await tree.root.findByType(TouchableOpacity).props.onPress();

    expect(open.mock.calls.length).toBe(1);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});