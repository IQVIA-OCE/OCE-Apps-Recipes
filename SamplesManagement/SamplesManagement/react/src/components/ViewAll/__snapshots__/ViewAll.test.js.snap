// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewAll Should render ViewAll component and handle press 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "borderTopColor": "#dddbda",
      "borderTopWidth": 1,
      "flexDirection": "row",
      "justifyContent": "flex-end",
      "marginTop": 5,
      "paddingTop": 10,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#0070d2",
          "fontSize": 13,
        }
      }
    >
      View all
    </Text>
  </View>
</View>
`;

exports[`ViewAll Should render ViewAll component and redirect to external url 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "borderTopColor": "#dddbda",
      "borderTopWidth": 1,
      "flexDirection": "row",
      "justifyContent": "flex-end",
      "marginTop": 5,
      "paddingTop": 10,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#0070d2",
          "fontSize": 13,
        }
      }
    >
      View all
    </Text>
  </View>
</View>
`;

exports[`ViewAll Should render ViewAll component and throw an error 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "borderTopColor": "#dddbda",
      "borderTopWidth": 1,
      "flexDirection": "row",
      "justifyContent": "flex-end",
      "marginTop": 5,
      "paddingTop": 10,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#0070d2",
          "fontSize": 13,
        }
      }
    >
      View all
    </Text>
  </View>
</View>
`;
